File: ChristmasTree/TreeCreator.cs
using System;

namespace ChristmasTree
{
    public class TreeCreator
    {
        protected int treeSize;
        protected bool hasPot;
        
        public TreeCreator()
        {
            Construct(10, true);
        }

        public TreeCreator(int size)
        {
            Construct(size, true);
        }
        
        public TreeCreator(int size, bool hasPot)
        {
            Construct(size, hasPot);
        }

        private void Construct(int size, bool hasPot)
        {
            this.treeSize = size;
            this.hasPot = hasPot;
        }
        
        public void CreateTree()
        {
            PrintTree();
            if (hasPot)
                PrintPlantPot();
            else
                PrintTrunk();
        }

        protected virtual void PrintTree()
        {
            Console.ForegroundColor = ConsoleColor.Green;
            int spaces = this.treeSize - 1;

            for (int depth = 0; depth < this.treeSize; depth++)
            {
                PrintSpaces(spaces);
                spaces--;
                
                PrintAsterisks(depth);
                Console.WriteLine();
            }
        }

        protected void PrintSpaces(int amount)
        {
            for (int spaces = 0; spaces < amount; spaces++)
            {
                Console.Write(" ");
            }
        }
        
        protected virtual void PrintAsterisks(int width)
        {
            PrintTreeHalf(width);
            PrintTreeHalf(width);
        }
        protected virtual void PrintTreeHalf(int width)
        {
            for (int counter = 0; counter <= width; counter++)
            {
                Console.ForegroundColor = ConsoleColor.Green;
                Console.Write("*");
            }
        }

        private void PrintPlantPot()
        {
            PrintSpaces(this.treeSize - 1);
            Console.ForegroundColor = ConsoleColor.Yellow;
            Console.WriteLine("**");
            PrintSpaces(this.treeSize - 3);
            Console.ForegroundColor = ConsoleColor.DarkMagenta;
            Console.WriteLine("******");
            PrintSpaces(this.treeSize - 3);
            Console.WriteLine("******");
        }
        
        private void PrintTrunk()
        {
            Console.ForegroundColor = ConsoleColor.Yellow;
            for (int count = 0; count < 3; count++)
            {
                PrintSpaces(this.treeSize - 1);
                Console.WriteLine("**");
            }
        }
    }
}

File: Python/TreeCreator.py
from termcolor import colored


class tree_creator:
    def __init__(self, size=10, has_pot=False):
        self.size = size
        self.has_pot = has_pot

    def create_tree(self):
        self.print_tree()
        if self.has_pot:
            self.print_pot()
        else:
            self.print_trunk()

    def print_tree(self):
        spaces = self.size

        for count in range(self.size + 1):
            self.print_spaces(spaces)
            spaces -= 1
            self.print_tree_left(count)
            self.print_tree_right(count)
            print()

    def print_tree_left(self, width):
        for count in range(width):
            print(colored("*", "green"), end="")

    def print_tree_right(self, width):
        for count in range(width):
            print(colored("*", "green"), end="")

    def print_spaces(self, amount):
        for count in range(amount):
            print(" ", end="")

    def print_pot(self):
        spaces = self.size - 1
        self.print_spaces(spaces)
        print(colored("**", "yellow"))

        spaces = self.size - 3
        self.print_spaces(spaces)
        print(colored("******", "magenta"))
        self.print_spaces(spaces)
        print(colored("******", "magenta"))

    def print_trunk(self):
        spaces = self.size - 1
        for count in range(3):
            self.print_spaces(spaces)
            print(colored("**", "yellow"))
